L1Trigger/L1THGCal/src/HGCalTriggerTowerGeometryHelper.cc

```diff
@@ -79,6 +80,129 @@ HGCalTriggerTowerGeometryHelper::HGCalTriggerTowerGeometryHelper(const edm::Para
     }
     l1tTriggerTowerMappingStream.close();
   }
+
+  if (splitModuleSum_) {
+    //variables for transforming towers
+    rotate180Deg_ = int(nBinsPhi_) / 2;
+    rotate120Deg_ = int(nBinsPhi_) / 3;
+    reverseX_ = int(nBinsPhi_) / 2 - 1;
+
+    std::ifstream moduleTowerMappingStream(conf.getParameter<edm::FileInPath>("moduleTowerMapping").fullPath());
+    if (!moduleTowerMappingStream.is_open()) {
+      throw cms::Exception("MissingDataFile") << "Cannot open HGCalTowerMapProducer moduleTowerMapping file\n";
+    }
+    //get split divisors
+    std::string line;
+    std::getline(moduleTowerMappingStream, line);  //Skip row
+    std::getline(moduleTowerMappingStream, line);
+    std::stringstream ss(line);
+    ss >> splitDivisorSilic_ >> splitDivisorScint_;
+
+    //get towers and module sum shares
+    std::getline(moduleTowerMappingStream, line);  //Skip row
+    std::getline(moduleTowerMappingStream, line);  //Skip row
+    const int minNumOfWordsPerRow = 5;
+    const int numOfWordsPerTower = 3;
+    for (std::string line; std::getline(moduleTowerMappingStream, line);) {
+      int numOfWordsInThisRow = 0;
+      for (std::string::size_type i = 0; i < line.size(); i++) {
+        if (line[i] != ' ' && line[i + 1] == ' ') {
+          numOfWordsInThisRow++;
+        }
+      }
+      if (numOfWordsInThisRow < minNumOfWordsPerRow) {
+        throw edm::Exception(edm::errors::Configuration, "Configuration")
+            << "HGCalTriggerTowerGeometryHelper warning: Incorrect/incomplete values for module ID in the mapping "
+               "file.\n"
+            << "The incorrect line is:" << line << std::endl;
+      }
+      int subdet;
+      int layer;
+      int moduleU;
+      int moduleV;
+      int numTowers;
```

Comment:
These should likely be initialized to some null value if possible

Reply:
Yes, indeed. Thanks @aloeliger 
Done